src/gurt/telemetry.c

```diff
@@ -2774,22 +2775,32 @@ d_tm_get_bucket_range(struct d_tm_context *ctx, struct d_tm_bucket_t *bucket,
  */
 int
 d_tm_get_counter(struct d_tm_context *ctx, uint64_t *val,
-		 struct d_tm_node_t *node)
+		 struct d_tm_node_t *node, bool for_srv)
 {
 	struct d_tm_metric_t	*metric_data = NULL;
 	struct d_tm_shmem_hdr	*shmem = NULL;
 	int			 rc;
 
-	if (ctx == NULL || val == NULL || node == NULL)
+	if (val == NULL || node == NULL)
+		return -DER_INVAL;
+
+	if (node->dtn_type != D_TM_COUNTER)
+		return -DER_OP_NOT_PERMITTED;
+
+	if (for_srv) {
```

Comment:
Any reason to not check for the ctx == NULL here instead, and remove the null check below? I would still prefer that over changing the API. If we add params to the function, there are changes needed in the control plane metrics collector as well.

Reply:
Just for the case that some client use case may call d_tm_get_counter() but miss to set the "ctx" parameter. Then it will misguide d_tm_get_counter() to regard it as a server side use case as to cause some unexpected behavior.